tests/network/sriov.go

```diff
@@ -520,6 +523,46 @@ var _ = Describe("[Serial]SRIOV", func() {
 			})
 		})
 
+		Context("Connected to multiple SRIOV networks", func() {
+			BeforeEach(func() {
+				Expect(createSriovNetworkAttachmentDefinition(sriovnet1, util.NamespaceTestDefault, sriovConfNAD)).To(Succeed(), shouldCreateNetwork)
+				Expect(createSriovNetworkAttachmentDefinition(sriovnet2, util.NamespaceTestDefault, sriovConfNAD)).To(Succeed(), shouldCreateNetwork)
+				Expect(createSriovNetworkAttachmentDefinition(sriovnet3, util.NamespaceTestDefault, sriovConfNAD)).To(Succeed(), shouldCreateNetwork)
+				Expect(createSriovNetworkAttachmentDefinition(sriovnet4, util.NamespaceTestDefault, sriovConfNAD)).To(Succeed(), shouldCreateNetwork)
+			})
+
+			It("should create a virtual machine with sriov interfaces with custom MAC addresses referring the appropriate resource", func() {
+				macAddressTemplate := "de:ad:00:be:ef:%02d"
+				pciAddressTemplate := "0000:%02d:00.0"
+				sriovNetworks := []string{sriovnet1, sriovnet2, sriovnet3, sriovnet4}
+				vmi := getSriovVmi(sriovNetworks, defaultCloudInitNetworkData())
+				for i := 1; i <= 4; i++ {
+					vmi.Spec.Domain.Devices.Interfaces[i].PciAddress = fmt.Sprintf(pciAddressTemplate, i)
+					vmi.Spec.Domain.Devices.Interfaces[i].MacAddress = fmt.Sprintf(macAddressTemplate, i)
+				}
+
+				vmi = startVmi(vmi)
+				vmi = waitVmi(vmi)
+
+				By("checking KUBEVIRT_RESOURCE_NAME_<networkName> variables are defined in pod")
+				for _, name := range sriovNetworks {
+					Expect(validatePodKubevirtResourceNameByVMI(virtClient, vmi, name, sriovResourceName)).To(Succeed())
+				}
+
+				checkDefaultInterfaceInPod(vmi)
+
+				By("checking virtual machine instance has three interfaces")
+				checkInterfacesInGuest(vmi, []string{"eth0", "eth1", "eth2", "eth3", "eth4"})
+
+				for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
+					assert.XFail("sriov interface mapping to pci-address is not working. Tracking issue - https://github.com/kubevirt/kubevirt/issues/5477", func() {
+						guestInterfaceName := getGuestInterfaceNameByMAC(vmi, iface.MacAddress)
```

Comment:
I agree with @oshoval that you should use `getInterfaceNameByMAC` instead.
It will also be correct to check it was found, because if it was not, the next command will blow up for the wrong reason.

Reply:
Ack, will also add an Eventually as the interfaceName does not appear at first

Reply:
this one Poll `findIfaceByMAC`